src/agent/src/windows_service.cpp

```diff
@@ -0,0 +1,176 @@
+#include <windows_service.hpp>
+
+#include <logger.hpp>
+
+#include <chrono>
+#include <thread>
+
+SERVICE_STATUS g_ServiceStatus = {};
+SERVICE_STATUS_HANDLE g_StatusHandle = nullptr;
+HANDLE g_ServiceStopEvent = INVALID_HANDLE_VALUE;
+
+namespace
+{
+    const std::string AGENT_SERVICENAME = "Wazuh Agent";
+
+    std::string GetExecutablePath()
+    {
+        char buffer[MAX_PATH];
+        GetModuleFileName(NULL, buffer, MAX_PATH);
+        return std::string(buffer);
+    }
+} // namespace
+
+namespace WindowsService
+{
+
+    void InstallService()
+    {
+        const std::string exePath = GetExecutablePath() + " service";
+
+        SC_HANDLE schSCManager = OpenSCManager(nullptr, nullptr, SC_MANAGER_CREATE_SERVICE);
+        if (schSCManager == nullptr)
+        {
+            LogError("OpenSCManager fail: {}", GetLastError());
+            return;
+        }
+
+        SC_HANDLE schService = CreateService(schSCManager,
+                                             AGENT_SERVICENAME.c_str(),
+                                             AGENT_SERVICENAME.c_str(),
+                                             SERVICE_ALL_ACCESS,
+                                             SERVICE_WIN32_OWN_PROCESS,
+                                             SERVICE_AUTO_START,
+                                             SERVICE_ERROR_NORMAL,
+                                             exePath.c_str(),
+                                             nullptr,
+                                             nullptr,
+                                             nullptr,
+                                             nullptr,
+                                             nullptr);
+
+        if (schService == nullptr)
+        {
+            LogError("CreateService fail: {}", GetLastError());
+            CloseServiceHandle(schSCManager);
+            return;
+        }
+
+        CloseServiceHandle(schService);
+        CloseServiceHandle(schSCManager);
+
+        LogInfo("Wazuh Agent Service successfully installed.");
+
+        return;
```

Comment:
This return is implied, same for lines 96, 97.

Reply:
No longer applies, as it now returns a bool!